larpandoracontent/LArThreeDReco/LArHitCreation/HitCreationBaseTool.cc

```diff
@@ -135,18 +169,28 @@ void HitCreationBaseTool::GetBestPosition3D(const HitType hitType, const Cartesi
 StatusCode HitCreationBaseTool::ReadSettings(const pandora::TiXmlHandle xmlHandle)
 {
     double sigmaX(std::sqrt(m_sigmaX2));
+    double sigmaYZ(std::sqrt(m_sigmaYZ2));
 
     PANDORA_RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::ReadValue(xmlHandle,
         "SigmaX", sigmaX));
+    PANDORA_RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::ReadValue(xmlHandle,
+        "SigmaYZ", sigmaYZ));
 
     m_sigmaX2 = sigmaX * sigmaX;
+    m_sigmaYZ2 = sigmaYZ * sigmaYZ;
 
```

Comment:
This section confuses me quite a bit.
 - `sigmaX` is constructed from a data member (`m_sigmaX2`) that is constructed in the initialised list.
 - `SigmaX` is then read in from an XML parameter.
 - `m_sigmaX2` is then set equal to the square of `sigmaX`

Couldn't `sigmaX` be the data member or could `sigmaX2` be read in from XML?
I realise the code you've added didn't set the precedent here but I feel like this should be discussed...

Reply:
I think there's a case for adding an m_sigmaX (etc) member variable to clean this up (I think we're stuck with the SigmaX XML parameter given existing usage). Doing it this way allows us to avoid the ugly sqrt before checking to see if an optional parameter has been provided and then recalculating the square from the input value. This was presumably done on efficiency grounds previously. So our options here would be to continue to have a squared member variable alongside a new m_sigmaX member, or dispense with the "efficiency" member variable, given that we actually have examples of products that could follow the same practise, but don't, elsewhere in this file.